providers/apis/defi/ethmulticlient/multi_client.go

```diff
@@ -0,0 +1,111 @@
+package ethmulticlient
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+	"go.uber.org/zap"
+
+	"github.com/skip-mev/slinky/oracle/config"
+)
+
+// MultiRPCClient implements the EVMClient interface by calling multiple underlying EVMClients and choosing
+// the best response.
+type MultiRPCClient struct {
+	clients   []EVMClient
+	endpoints []config.Endpoint
+	logger    *zap.Logger
+}
+
+// NewMultiRPCClient returns a new MultiRPCClient.
+func NewMultiRPCClient(clients []EVMClient, endpoints []config.Endpoint, logger *zap.Logger) *MultiRPCClient {
+	return &MultiRPCClient{
+		clients:   clients,
+		endpoints: endpoints,
+		logger:    logger,
+	}
+}
+
+// NewMultiRPCClientFromEndpoints creates a MultiRPCClient from config endpoints.
+func NewMultiRPCClientFromEndpoints(ctx context.Context, logger *zap.Logger, endpoints []config.Endpoint) (*MultiRPCClient, error) {
+	if logger == nil {
+		return nil, fmt.Errorf("logger cannot be nil")
+	}
+	clients := make([]EVMClient, len(endpoints))
+	for i, endpoint := range endpoints {
+		var err error
+		clients[i], err = NewGoEthereumClientImplFromEndpoint(ctx, endpoint)
+		if err != nil {
+			logger.Error(
```

Comment:
do we need this log?

Reply:
This should only be called on provider initialization so it seems ok to log this when we fail to initialize the provider?